### PR TITLE
[libs][Android] Filter out backwards timezones only with valid tzlookup file

### PR DESCRIPTION
**Pull request by mdh1418**
Partial fix for https://github.com/dotnet/runtime/issues/81373

<details>
<summary>The resulting collection of unfiltered timezone ids</summary>
TimeZone: (UTC-12:00) GMT-12:00
TimeZone: (UTC-11:00) GMT-11:00
TimeZone: (UTC-11:00) Niue Time
TimeZone: (UTC-11:00) Samoa Standard Time (Midway)
TimeZone: (UTC-11:00) Samoa Standard Time (Pago Pago)
TimeZone: (UTC-11:00) Samoa Standard Time (Pago Pago)
TimeZone: (UTC-11:00) Samoa Standard Time (Pago Pago)
TimeZone: (UTC-10:00) Cook Islands Standard Time (Rarotonga)
TimeZone: (UTC-10:00) GMT-10:00
TimeZone: (UTC-10:00) GMT-10:00
TimeZone: (UTC-10:00) Hawaii-Aleutian Time (Adak)
TimeZone: (UTC-10:00) Hawaii-Aleutian Time (Adak)
TimeZone: (UTC-10:00) Hawaii-Aleutian Time (Adak)
TimeZone: (UTC-10:00) Hawaii-Aleutian Time (Adak) (Honolulu)
TimeZone: (UTC-10:00) Hawaii-Aleutian Time (Adak) (Honolulu)
TimeZone: (UTC-10:00) Hawaii-Aleutian Time (Adak) (Johnston)
TimeZone: (UTC-10:00) Tahiti Time
TimeZone: (UTC-09:30) Marquesas Time
TimeZone: (UTC-09:00) Alaska Time (Anchorage)
TimeZone: (UTC-09:00) Alaska Time (Anchorage)
TimeZone: (UTC-09:00) Alaska Time (Juneau)
TimeZone: (UTC-09:00) Alaska Time (Metlakatla)
TimeZone: (UTC-09:00) Alaska Time (Nome)
TimeZone: (UTC-09:00) Alaska Time (Sitka)
TimeZone: (UTC-09:00) Alaska Time (Yakutat)
TimeZone: (UTC-09:00) GMT-09:00
TimeZone: (UTC-09:00) Gambier Time
TimeZone: (UTC-08:00) GMT-08:00
TimeZone: (UTC-08:00) Northwest Mexico Time (Santa Isabel)
TimeZone: (UTC-08:00) Pacific Time (Dawson)
TimeZone: (UTC-08:00) Pacific Time (Los Angeles)
TimeZone: (UTC-08:00) Pacific Time (Los Angeles)
TimeZone: (UTC-08:00) Pacific Time (Tijuana)
TimeZone: (UTC-08:00) Pacific Time (Tijuana)
TimeZone: (UTC-08:00) Pacific Time (Tijuana)
TimeZone: (UTC-08:00) Pacific Time (Unknown City)
TimeZone: (UTC-08:00) Pacific Time (Vancouver)
TimeZone: (UTC-08:00) Pacific Time (Vancouver)
TimeZone: (UTC-08:00) Pacific Time (Whitehorse)
TimeZone: (UTC-08:00) Pacific Time (Whitehorse)
TimeZone: (UTC-08:00) Pitcairn Islands Time
TimeZone: (UTC-07:00) GMT-07:00
TimeZone: (UTC-07:00) GMT-07:00
TimeZone: (UTC-07:00) Mexican Pacific Time (Chihuahua)
TimeZone: (UTC-07:00) Mexican Pacific Time (Hermosillo)
TimeZone: (UTC-07:00) Mexican Pacific Time (Mazatlan)
TimeZone: (UTC-07:00) Mexican Pacific Time (Mazatlan)
TimeZone: (UTC-07:00) Mountain Time (Boise)
TimeZone: (UTC-07:00) Mountain Time (Cambridge Bay)
TimeZone: (UTC-07:00) Mountain Time (Creston)
TimeZone: (UTC-07:00) Mountain Time (Dawson Creek)
TimeZone: (UTC-07:00) Mountain Time (Denver)
TimeZone: (UTC-07:00) Mountain Time (Denver)
TimeZone: (UTC-07:00) Mountain Time (Denver)
TimeZone: (UTC-07:00) Mountain Time (Denver)
TimeZone: (UTC-07:00) Mountain Time (Edmonton)
TimeZone: (UTC-07:00) Mountain Time (Edmonton)
TimeZone: (UTC-07:00) Mountain Time (Fort Nelson)
TimeZone: (UTC-07:00) Mountain Time (Inuvik)
TimeZone: (UTC-07:00) Mountain Time (Ojinaga)
TimeZone: (UTC-07:00) Mountain Time (Phoenix)
TimeZone: (UTC-07:00) Mountain Time (Phoenix)
TimeZone: (UTC-07:00) Mountain Time (Unknown City)
TimeZone: (UTC-07:00) Mountain Time (Yellowknife)
TimeZone: (UTC-06:00) Central Time (Bahia Banderas)
TimeZone: (UTC-06:00) Central Time (Belize)
TimeZone: (UTC-06:00) Central Time (Beulah, North Dakota)
TimeZone: (UTC-06:00) Central Time (Center, North Dakota)
TimeZone: (UTC-06:00) Central Time (Chicago)
TimeZone: (UTC-06:00) Central Time (Chicago)
TimeZone: (UTC-06:00) Central Time (Costa Rica)
TimeZone: (UTC-06:00) Central Time (El Salvador)
TimeZone: (UTC-06:00) Central Time (Guatemala)
TimeZone: (UTC-06:00) Central Time (Knox, Indiana)
TimeZone: (UTC-06:00) Central Time (Knox, Indiana)
TimeZone: (UTC-06:00) Central Time (Knox, Indiana)
TimeZone: (UTC-06:00) Central Time (Managua)
TimeZone: (UTC-06:00) Central Time (Matamoros)
TimeZone: (UTC-06:00) Central Time (Menominee)
TimeZone: (UTC-06:00) Central Time (Merida)
TimeZone: (UTC-06:00) Central Time (Mexico City)
TimeZone: (UTC-06:00) Central Time (Mexico City)
TimeZone: (UTC-06:00) Central Time (Monterrey)
TimeZone: (UTC-06:00) Central Time (New Salem, North Dakota)
TimeZone: (UTC-06:00) Central Time (Rainy River)
TimeZone: (UTC-06:00) Central Time (Rankin Inlet)
TimeZone: (UTC-06:00) Central Time (Regina)
TimeZone: (UTC-06:00) Central Time (Regina)
TimeZone: (UTC-06:00) Central Time (Regina)
TimeZone: (UTC-06:00) Central Time (Resolute)
TimeZone: (UTC-06:00) Central Time (Swift Current)
TimeZone: (UTC-06:00) Central Time (Tegucigalpa)
TimeZone: (UTC-06:00) Central Time (Tell City, Indiana)
TimeZone: (UTC-06:00) Central Time (Unknown City)
TimeZone: (UTC-06:00) Central Time (Winnipeg)
TimeZone: (UTC-06:00) Central Time (Winnipeg)
TimeZone: (UTC-06:00) Easter Island Time
TimeZone: (UTC-06:00) Easter Island Time
TimeZone: (UTC-06:00) GMT-06:00
TimeZone: (UTC-06:00) Galapagos Time
TimeZone: (UTC-05:00) Acre Standard Time (Eirunepe)
TimeZone: (UTC-05:00) Acre Standard Time (Rio Branco)
TimeZone: (UTC-05:00) Acre Standard Time (Rio Branco)
TimeZone: (UTC-05:00) Acre Standard Time (Rio Branco)
TimeZone: (UTC-05:00) Colombia Standard Time (Bogota)
TimeZone: (UTC-05:00) Cuba Time
TimeZone: (UTC-05:00) Cuba Time
TimeZone: (UTC-05:00) Eastern Time (Atikokan)
TimeZone: (UTC-05:00) Eastern Time (Atikokan)
TimeZone: (UTC-05:00) Eastern Time (Cancun)
TimeZone: (UTC-05:00) Eastern Time (Cayman)
TimeZone: (UTC-05:00) Eastern Time (Detroit)
TimeZone: (UTC-05:00) Eastern Time (Detroit)
TimeZone: (UTC-05:00) Eastern Time (Indianapolis)
TimeZone: (UTC-05:00) Eastern Time (Indianapolis)
TimeZone: (UTC-05:00) Eastern Time (Indianapolis)
TimeZone: (UTC-05:00) Eastern Time (Indianapolis)
TimeZone: (UTC-05:00) Eastern Time (Iqaluit)
TimeZone: (UTC-05:00) Eastern Time (Jamaica)
TimeZone: (UTC-05:00) Eastern Time (Jamaica)
TimeZone: (UTC-05:00) Eastern Time (Louisville)
TimeZone: (UTC-05:00) Eastern Time (Louisville)
TimeZone: (UTC-05:00) Eastern Time (Marengo, Indiana)
TimeZone: (UTC-05:00) Eastern Time (Monticello, Kentucky)
TimeZone: (UTC-05:00) Eastern Time (Nassau)
TimeZone: (UTC-05:00) Eastern Time (New York)
TimeZone: (UTC-05:00) Eastern Time (New York)
TimeZone: (UTC-05:00) Eastern Time (Nipigon)
TimeZone: (UTC-05:00) Eastern Time (Panama)
TimeZone: (UTC-05:00) Eastern Time (Pangnirtung)
TimeZone: (UTC-05:00) Eastern Time (Petersburg, Indiana)
TimeZone: (UTC-05:00) Eastern Time (Port-au-Prince)
TimeZone: (UTC-05:00) Eastern Time (Thunder Bay)
TimeZone: (UTC-05:00) Eastern Time (Toronto)
TimeZone: (UTC-05:00) Eastern Time (Toronto)
TimeZone: (UTC-05:00) Eastern Time (Unknown City)
TimeZone: (UTC-05:00) Eastern Time (Vevay, Indiana)
TimeZone: (UTC-05:00) Eastern Time (Vincennes, Indiana)
TimeZone: (UTC-05:00) Eastern Time (Winamac, Indiana)
TimeZone: (UTC-05:00) Ecuador Time
TimeZone: (UTC-05:00) GMT-05:00
TimeZone: (UTC-05:00) GMT-05:00
TimeZone: (UTC-05:00) Montreal Time
TimeZone: (UTC-05:00) Peru Standard Time (Lima)
TimeZone: (UTC-04:00) Amazon Time (Boa Vista)
TimeZone: (UTC-04:00) Amazon Time (Campo Grande)
TimeZone: (UTC-04:00) Amazon Time (Cuiaba)
TimeZone: (UTC-04:00) Amazon Time (Manaus)
TimeZone: (UTC-04:00) Amazon Time (Manaus)
TimeZone: (UTC-04:00) Amazon Time (Porto Velho)
TimeZone: (UTC-04:00) Atlantic Time (Anguilla)
TimeZone: (UTC-04:00) Atlantic Time (Antigua)
TimeZone: (UTC-04:00) Atlantic Time (Aruba)
TimeZone: (UTC-04:00) Atlantic Time (Barbados)
TimeZone: (UTC-04:00) Atlantic Time (Bermuda)
TimeZone: (UTC-04:00) Atlantic Time (Blanc-Sablon)
TimeZone: (UTC-04:00) Atlantic Time (Curaçao)
TimeZone: (UTC-04:00) Atlantic Time (Dominica)
TimeZone: (UTC-04:00) Atlantic Time (Glace Bay)
TimeZone: (UTC-04:00) Atlantic Time (Goose Bay)
TimeZone: (UTC-04:00) Atlantic Time (Grand Turk)
TimeZone: (UTC-04:00) Atlantic Time (Grenada)
TimeZone: (UTC-04:00) Atlantic Time (Guadeloupe)
TimeZone: (UTC-04:00) Atlantic Time (Halifax)
TimeZone: (UTC-04:00) Atlantic Time (Halifax)
TimeZone: (UTC-04:00) Atlantic Time (Kralendijk)
TimeZone: (UTC-04:00) Atlantic Time (Lower Prince’s Quarter)
TimeZone: (UTC-04:00) Atlantic Time (Marigot)
TimeZone: (UTC-04:00) Atlantic Time (Martinique)
TimeZone: (UTC-04:00) Atlantic Time (Moncton)
TimeZone: (UTC-04:00) Atlantic Time (Montserrat)
TimeZone: (UTC-04:00) Atlantic Time (Port of Spain)
TimeZone: (UTC-04:00) Atlantic Time (Puerto Rico)
TimeZone: (UTC-04:00) Atlantic Time (Santo Domingo)
TimeZone: (UTC-04:00) Atlantic Time (St. Barthélemy)
TimeZone: (UTC-04:00) Atlantic Time (St. Kitts)
TimeZone: (UTC-04:00) Atlantic Time (St. Lucia)
TimeZone: (UTC-04:00) Atlantic Time (St. Thomas)
TimeZone: (UTC-04:00) Atlantic Time (St. Thomas)
TimeZone: (UTC-04:00) Atlantic Time (St. Vincent)
TimeZone: (UTC-04:00) Atlantic Time (Thule)
TimeZone: (UTC-04:00) Atlantic Time (Tortola)
TimeZone: (UTC-04:00) Bolivia Time
TimeZone: (UTC-04:00) Chile Time
TimeZone: (UTC-04:00) Chile Time
TimeZone: (UTC-04:00) Chile Time (Palmer)
TimeZone: (UTC-04:00) GMT-04:00
TimeZone: (UTC-04:00) Guyana Time
TimeZone: (UTC-04:00) Paraguay Time
TimeZone: (UTC-04:00) Venezuela Time
TimeZone: (UTC-03:30) Newfoundland Time (St. John’s)
TimeZone: (UTC-03:30) Newfoundland Time (St. John’s)
TimeZone: (UTC-03:00) Argentina Standard Time (Buenos Aires)
TimeZone: (UTC-03:00) Argentina Standard Time (Buenos Aires)
TimeZone: (UTC-03:00) Argentina Standard Time (Catamarca)
TimeZone: (UTC-03:00) Argentina Standard Time (Catamarca)
TimeZone: (UTC-03:00) Argentina Standard Time (Catamarca)
TimeZone: (UTC-03:00) Argentina Standard Time (Cordoba)
TimeZone: (UTC-03:00) Argentina Standard Time (Cordoba)
TimeZone: (UTC-03:00) Argentina Standard Time (Cordoba)
TimeZone: (UTC-03:00) Argentina Standard Time (Jujuy)
TimeZone: (UTC-03:00) Argentina Standard Time (Jujuy)
TimeZone: (UTC-03:00) Argentina Standard Time (La Rioja)
TimeZone: (UTC-03:00) Argentina Standard Time (Mendoza)
TimeZone: (UTC-03:00) Argentina Standard Time (Mendoza)
TimeZone: (UTC-03:00) Argentina Standard Time (Rio Gallegos)
TimeZone: (UTC-03:00) Argentina Standard Time (Salta)
TimeZone: (UTC-03:00) Argentina Standard Time (San Juan)
TimeZone: (UTC-03:00) Argentina Standard Time (Tucuman)
TimeZone: (UTC-03:00) Argentina Standard Time (Ushuaia)
TimeZone: (UTC-03:00) Brasilia Time (Araguaina)
TimeZone: (UTC-03:00) Brasilia Time (Bahia)
TimeZone: (UTC-03:00) Brasilia Time (Belem)
TimeZone: (UTC-03:00) Brasilia Time (Fortaleza)
TimeZone: (UTC-03:00) Brasilia Time (Maceio)
TimeZone: (UTC-03:00) Brasilia Time (Recife)
TimeZone: (UTC-03:00) Brasilia Time (Santarem)
TimeZone: (UTC-03:00) Brasilia Time (Sao Paulo)
TimeZone: (UTC-03:00) Brasilia Time (Sao Paulo)
TimeZone: (UTC-03:00) Falkland Islands Standard Time (Stanley)
TimeZone: (UTC-03:00) French Guiana Time
TimeZone: (UTC-03:00) GMT-03:00
TimeZone: (UTC-03:00) Rothera Time
TimeZone: (UTC-03:00) St. Pierre & Miquelon Time
TimeZone: (UTC-03:00) Suriname Time
TimeZone: (UTC-03:00) Uruguay Standard Time (Montevideo)
TimeZone: (UTC-03:00) West Greenland Time (Nuuk)
TimeZone: (UTC-03:00) Western Argentina Standard Time (San Luis)
TimeZone: (UTC-02:00) Fernando de Noronha Standard Time
TimeZone: (UTC-02:00) Fernando de Noronha Standard Time
TimeZone: (UTC-02:00) GMT-02:00
TimeZone: (UTC-02:00) South Georgia Time
TimeZone: (UTC-01:00) Azores Time
TimeZone: (UTC-01:00) Cape Verde Standard Time
TimeZone: (UTC-01:00) East Greenland Time (Ittoqqortoormiit)
TimeZone: (UTC-01:00) GMT-01:00
TimeZone: (UTC) Coordinated Universal Time
TimeZone: (UTC) Coordinated Universal Time
TimeZone: (UTC) Coordinated Universal Time
TimeZone: (UTC) Coordinated Universal Time
TimeZone: (UTC) Coordinated Universal Time
TimeZone: (UTC) Coordinated Universal Time
TimeZone: (UTC) Coordinated Universal Time
TimeZone: (UTC) Coordinated Universal Time
TimeZone: (UTC+00:00) GMT
TimeZone: (UTC+00:00) GMT
TimeZone: (UTC+00:00) GMT
TimeZone: (UTC+00:00) GMT
TimeZone: (UTC+00:00) GMT
TimeZone: (UTC+00:00) GMT
TimeZone: (UTC+00:00) GMT+01:00
TimeZone: (UTC+00:00) Greenwich Mean Time (Abidjan)
TimeZone: (UTC+00:00) Greenwich Mean Time (Accra)
TimeZone: (UTC+00:00) Greenwich Mean Time (Bamako)
TimeZone: (UTC+00:00) Greenwich Mean Time (Bamako)
TimeZone: (UTC+00:00) Greenwich Mean Time (Banjul)
TimeZone: (UTC+00:00) Greenwich Mean Time (Bissau)
TimeZone: (UTC+00:00) Greenwich Mean Time (Conakry)
TimeZone: (UTC+00:00) Greenwich Mean Time (Dakar)
TimeZone: (UTC+00:00) Greenwich Mean Time (Danmarkshavn)
TimeZone: (UTC+00:00) Greenwich Mean Time (Freetown)
TimeZone: (UTC+00:00) Greenwich Mean Time (Lome)
TimeZone: (UTC+00:00) Greenwich Mean Time (Monrovia)
TimeZone: (UTC+00:00) Greenwich Mean Time (Nouakchott)
TimeZone: (UTC+00:00) Greenwich Mean Time (Ouagadougou)
TimeZone: (UTC+00:00) Greenwich Mean Time (Reykjavik)
TimeZone: (UTC+00:00) Greenwich Mean Time (Reykjavik)
TimeZone: (UTC+00:00) Greenwich Mean Time (St. Helena)
TimeZone: (UTC+00:00) Greenwich Mean Time (São Tomé)
TimeZone: (UTC+00:00) Guernsey Time
TimeZone: (UTC+00:00) Ireland Time
TimeZone: (UTC+00:00) Ireland Time
TimeZone: (UTC+00:00) Isle of Man Time
TimeZone: (UTC+00:00) Jersey Time
TimeZone: (UTC+00:00) Troll Time
TimeZone: (UTC+00:00) United Kingdom Time
TimeZone: (UTC+00:00) United Kingdom Time
TimeZone: (UTC+00:00) United Kingdom Time

</details>

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs [hunk 1]
```diff
@@ -265,9 +265,13 @@ public AndroidTzData()
             //
             // Once the timezone cache is populated with the IDs, we reference tzlookup id tags
             // to determine if an id is backwards and label it as such if they are.
-            private static void FilterBackwardIDs(string tzFileDir, out HashSet<string> tzLookupIDs)
+            private static bool FilterBackwardIDs(string tzFileDir, out HashSet<string> tzLookupIDs)
             {
                 tzLookupIDs = new HashSet<string>();
```
Comment (tarekgh) [inline src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs:270]: maybe delay allocation after the file exist check?

**Review by tarekgh — APPROVED** [review]
Add a few suggestions. LGTM otherwise.

File: src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs [hunk 2]
```diff
@@ -265,9 +265,13 @@ public AndroidTzData()
...
+                string tzFilePath = Path.Combine(tzFileDir, "tzlookup.xml");
+                if (!File.Exists(tzFilePath))
+                    return false;
+
                 try
                 {
                     using (StreamReader sr = File.OpenText(Path.Combine(tzFileDir, "tzlookup.xml")))
```
[the remark this answers is not in the document]
Comment (mdh1418) [reply src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs]: Utilized variable, updated variable name to be `tzLookupFilePath` to not confuse with `tzFilePath` previously used to allude to the `tzdata` file.

File: src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs (hunk 1 above)
Comment (mdh1418) [reply src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs]: Done, thanks for the suggestion!

File: src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs [hunk 3]
```diff
@@ -265,9 +265,13 @@ public AndroidTzData()
             //
             // Once the timezone cache is populated with the IDs, we reference tzlookup id tags
             // to determine if an id is backwards and label it as such if they are.
-            private static void FilterBackwardIDs(string tzFileDir, out HashSet<string> tzLookupIDs)
+            private static bool FilterBackwardIDs(string tzFileDir, out HashSet<string> tzLookupIDs)
```
[the remark this answers is not in the document]
Comment (tarekgh) [reply src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs]: fine with me.

**Review by simonrozsival — APPROVED** [review]

**Comment by mdh1418** [comment]
Need to modify the test `NoBackwardTimeZones`

**Pull request merged**